### ISSUE TITLE
UseReleaseStack does not work with ActiveX

### ISSUE THREAD

**Issue by ericbonilha** [pleriche/FastMM4#39]
Hello

I'm using latest 4992 version and I activated UseReleaseStack option because our application is highly, and I mean, HIGHLY multi-threaded and we were suffering a lot with thread contention.

I tried a different memory manager like ScaleMM2 and performance went to the roof, great, but somehow that library uses too much memory and we decided to go back to FastMM and we found this new flag which dramatically reduced thread contention and decreased about 30 to 40% of CPU usage from what we used to have...

The problem we are seeing now is our ActiveX controls, if we use UseReleaseStack flag, the ActiveX control does not work, it hangs... even Delphi cannot import a unit with the Interface of the activeX control (once its compiled)

If we register the control with regsrv32, regsrv32 will register it and hang after registering.

Is there a reason for the ActiveX control not work with this flag activated?

Meanwhile we will remove FastMM from our ActiveX projects because we need UseReleaseStack for our software, it makes wonders!

**Comment by pleriche** [comment]
Hi Eric,

I don't have any bug reports for the UseReleaseStack feature, but it is fairly new so there is always the possibility that there is some rare race condition or other bug that has not been discovered yet. If you can get a small test case together I will take a look. My e-mail is in the source files.

Best regards,
Pierre

**Comment by maximmasiutin** [comment]
You are right about LongInt being 64-bit on Delphi Linux 64-bit (LP64 data model). This is the root cause.

TSynchronizationVariable was defined as LongInt when compiled for Delphi Linux. On LP64, LongInt is 8 bytes, which caused AtomicExchange to use the Int64 overload (8-byte atomic exchange instead of 4-byte), corrupting adjacent struct fields and triggering _BoundErr when the result was assigned to a Byte variable.

Fix in PR #48: use Integer (always 4 bytes in Delphi) instead of LongInt for the Delphi path, keep LongInt for FPC (where it is always 4 bytes). Added a compile-time size guard.
